lib/repl.js

```diff
@@ -534,9 +544,50 @@ function REPLServer(prompt,
   });
 
   self.on('SIGCONT', function() {
-    self.displayPrompt(true);
+    if (self.editorMode) {
+      self.outputStream.write(`${self._initialPrompt}.editor\n`);
+      self.outputStream.write(
+        '// Entering editor mode (^D to finish, ^C to cancel)\n');
+      self.outputStream.write(`${self.bufferedCommand}\n`);
+      self.prompt(true);
+    } else {
+      self.displayPrompt(true);
+    }
   });
 
+  // Wrap readline tty to enable editor mode
+  const ttyWrite = self._ttyWrite.bind(self);
```

Comment:
Looks like I am totally missing something here. Why `_ttyWrite` needs to be bound to `self`?